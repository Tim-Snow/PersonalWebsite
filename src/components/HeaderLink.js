import React from 'react'

const HeaderLink = ({name}) => (
    <div style={styles.container}>
        {name}
    </div>
)

const styles = {
    container: {
        backgroundColor: '#fff', 
        margin: 4, 
        padding: 3, 
        borderRadius: 5, 
        border: '1px solid black', 
        textAlign: 'center', 
        display: 'block'
    }
}

export default HeaderLink
