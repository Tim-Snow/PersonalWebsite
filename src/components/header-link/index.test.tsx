import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import HeaderLink from './index'

configure({ adapter: new Adapter() })

const defaultProps = {
  cb: () => undefined,
  text: 'a',
}

describe('header link', () => {
  it('should match the snapshot', () => {
    const headerlink = shallow(<HeaderLink {...defaultProps} />)
    expect(headerlink).toMatchSnapshot()
  })
})
