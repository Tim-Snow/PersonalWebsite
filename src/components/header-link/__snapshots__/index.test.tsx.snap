// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header link should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderLink
    cb={[Function]}
    text="a"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "a",
      "onClick": [Function],
      "onMouseLeave": [Function],
      "onMouseOver": [Function],
      "style": Object {
        "cursor": "pointer",
        "flexGrow": 0.5,
      },
    },
    "ref": null,
    "rendered": "a",
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "a",
        "onClick": [Function],
        "onMouseLeave": [Function],
        "onMouseOver": [Function],
        "style": Object {
          "cursor": "pointer",
          "flexGrow": 0.5,
        },
      },
      "ref": null,
      "rendered": "a",
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
